test/validation/api/scheduler/scheduler.c

```diff
@@ -565,6 +565,104 @@ static void scheduler_test_queue_size(void)
 	CU_ASSERT_FATAL(odp_pool_destroy(pool) == 0);
 }
 
+static void test_queue_priority(odp_schedule_sync_t sync, odp_bool_t mixed_sync)
+{
+	odp_schedule_capability_t sched_capa;
+	odp_schedule_prio_t prev_prio;
+	odp_queue_param_t queue_param;
+	const int max_prios = 128;
+	odp_queue_t queue[max_prios];
+	uint64_t wait_time;
+	const uint32_t event_per_prio = 4;
+	uint32_t num_events_prio[max_prios];
+	uint32_t tot_events;
+	uint32_t num_events = 0;
+	int num_prios = ODPH_MIN(odp_schedule_num_prio(), max_prios);
+	odp_schedule_sync_t sync_tbl[] = {ODP_SCHED_SYNC_PARALLEL,
+					  ODP_SCHED_SYNC_ATOMIC,
+					  ODP_SCHED_SYNC_ORDERED};
+
+	memset(num_events_prio, 0, max_prios * sizeof(uint32_t));
+
+	CU_ASSERT_FATAL(odp_schedule_capability(&sched_capa) == 0);
+	if ((uint32_t)num_prios > sched_capa.max_queues)
+		num_prios = sched_capa.max_queues;
+	tot_events = event_per_prio * num_prios;
+
+	sched_queue_param_init(&queue_param);
+	queue_param.sched.sync = sync;
+
+	/* Try to prevent possible event pre-scheduling */
+	odp_schedule_pause();
```

Comment:
I guess for extra safety, `drain_queues()` could be called here also.

Reply:
I did not add extra drain here, reason being it's the responsibility of each test to clean up the queues afterwards and `drain_queues()` takes some time.